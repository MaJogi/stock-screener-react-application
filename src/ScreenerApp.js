import React from "react";
import {BrowserRouter as Router, Route, Switch} from "react-router-dom"
import CompanyInfoComponent from "./components/CompanyInfoComponent";
import TableTabsComponent from "./components/TableTabsComponent";

class ScreenerApp extends React.Component {

    render() {
        return (
            <Router history={History}>
                <div>
                    <Switch>
                        <Route exact path={"/"} component={TableTabsComponent}/>
                        <Route path={"/ticker/:id"} component={CompanyInfoComponent}/>
                    </Switch>
                </div>
            </Router>
        )
    }
}

export default ScreenerApp