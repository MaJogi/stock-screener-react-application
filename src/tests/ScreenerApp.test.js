import React from "react";
import {cleanup, render} from '@testing-library/react'
import ScreenerApp from "../ScreenerApp";

afterEach(cleanup)

it('should take a snapshot', () => {
    const {asFragment} = render(<ScreenerApp/>)

    expect(asFragment(<ScreenerApp/>)).toMatchSnapshot()
});