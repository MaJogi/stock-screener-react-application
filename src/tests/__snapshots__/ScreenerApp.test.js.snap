// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="p-menubar p-component"
    >
      <ul
        class="p-menubar-root-list"
        role="menubar"
      >
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="false"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-home"
            />
            <span
              class="p-menuitem-text"
            >
              Home
            </span>
          </a>
        </li>
        <li
          class="p-menuitem"
          role="none"
        >
          <a
            aria-haspopup="false"
            class="p-menuitem-link"
            href="#"
            role="menuitem"
          >
            <span
              class="p-menuitem-icon pi pi-desktop"
            />
            <span
              class="p-menuitem-text"
            >
              Screener
            </span>
          </a>
        </li>
      </ul>
    </div>
    <div
      style="position: relative; height: 100vh;"
    >
      <img
        alt="Stock Market"
        src="StockMarket.PNG"
        style="position: absolute; width: 100%; height: 50%; opacity: 0.35; z-index: -1;"
      />
      <div
        style="width: 100%; height: 50%; text-align: center;"
      >
        <h1
          style="margin: auto; padding-top: 10vh; width: 100vw; font-size: 8vh;"
        >
          Welcome to FinTrust!
        </h1>
        <h2
          style="margin: auto; font-size: 3.5vh;"
        >
          Your guide to the Tallinn Stock Exchange.
        </h2>
      </div>
      <div
        style="width: 100%; height: 50%;"
      >
        <div>
          <div />
        </div>
        <div
          style="position: fixed; left: 50%; transform: translate(-50%);"
        >
          <div
            class="p-inputgroup"
          >
            <span
              aria-haspopup="listbox"
              aria-owns="pr_id_1_list"
              class="p-autocomplete p-component"
            >
              <input
                aria-autocomplete="list"
                aria-controls="pr_id_1_list"
                autocomplete="off"
                class="p-inputtext p-component p-autocomplete-input"
                placeholder="Search for tickers"
                role="searchbox"
                type="text"
                value=""
              />
              <i
                class="p-autocomplete-loader pi pi-spinner pi-spin"
                style="visibility: hidden;"
              />
              <div
                class="p-autocomplete-panel p-input-overlay"
                style="max-height: 200px;"
              >
                <ul
                  class="p-autocomplete-items p-autocomplete-list p-component p-reset"
                  id="pr_id_1_list"
                  role="listbox"
                />
              </div>
            </span>
            <button
              class="p-button p-component p-button-icon-only"
            >
              <span
                class="pi pi-search p-c p-button-icon-left"
              />
              <span
                class="p-button-text p-c"
              >
                p-btn
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
